contentcuration/contentcuration/assets/channelList/views/Channel/ChannelList.vue

```diff
@@ -0,0 +1,149 @@
+      <VFlex xs12 sm10 md6>
+        <VLayout row wrap justify-end>
+          <VFlex xs2>
+            <VBtn v-if="isEditable" color="primary" @click="newChannel">
+              <VIcon>add</VIcon>
+              {{ $tr('channel') }}
+            </VBtn>
+          </VFlex>
+        </VLayout>
+        <VLayout row justify-center>
+          <VFlex xs12>
+            <template v-if="loading">
+              <VProgressLinear
+                indeterminate
+                color="primary"
+              />
+              <p class="headline mb-0">
+                {{ $tr('loading') }}
+              </p>
+            </template>
+            <p v-else-if="listChannels && !listChannels.length" class="headline mb-0">
+              {{ $tr('noChannelsFound') }}
+            </p>
+            <template v-else>
+              <ChannelItem
+                v-for="channel in listChannels"
+                :key="channel.id"
+                :channel="channel"
+              />
+            </template>
+            <router-view v-if="!loading"/>
+          </VFlex>
+        </VLayout>
+      </VFlex>
+    </VLayout>
+  </VContainer>
+</template>
+
+
+<script>
+
+  import sortBy from 'lodash/sortBy';
+  import { mapGetters, mapActions, mapState, mapMutations } from 'vuex';
+  import { ListTypes, RouterNames } from '../../constants';
+  import ChannelItem from './ChannelItem.vue';
+  import { generateTempId, isTempId } from '../../utils';
+
+  function listTypeValidator(value) {
+    // The value must match one of the ListTypes
+    return Object.values(ListTypes).includes(value);
+  }
+
+  export default {
+    name: 'ChannelList',
+    $trs: {
+      loading: 'Loading channels...',
+      noChannelsFound: 'No channels found',
+      channel: 'Channel',
+      addChannel: 'Create a new channel',
+    },
+    components: {
+      ChannelItem,
+    },
+    props: {
+      listType: {
+        type: String,
+        validator: listTypeValidator,
+      },
+    },
+    data() {
+      return {
+        loading: true,
+        newSetId: generateTempId(),
+      };
+    },
+    computed: {
+      ...mapGetters('channelList', ['channels']),
+      ...mapState({
+        language: state => state.session.currentLanguage,
+        preferences: state => state.session.preferences,
+      }),
+      listChannels() {
+        const sortFields = ['-modified'];
+        if (this.listType === ListTypes.PUBLIC) {
+          sortFields.shift('-priority');
+        }
+        return sortBy(this.channels.filter(channel => channel[this.listType]), sortFields);
+      },
+      isEditable() {
+        return this.listType === ListTypes.EDITABLE;
+      },
+    },
+    beforeRouteEnter(to, from, next) {
+      if(listTypeValidator(to.params.listType)) {
+        return next(vm => {
+          vm.loadData(to.params.listType);
+        });
+      }
+      return next(false);
+    },
+    beforeRouteUpdate(to, from, next) {
+      if(listTypeValidator(to.params.listType)) {
```

Comment:
To check: how does the catch all route interact with this?

Reply:
This seems to be fine, because the catch all route just forwards to the channel edit list.